javalib/src/main/scala/java/net/InetAddress.scala

```diff
@@ -0,0 +1,648 @@
+      }
+    }
+    return false
+  }
+
+  def isLinkLocalAddress(): Boolean
+
+  def isAnyLocalAddress(): Boolean
+
+  def isLoopbackAddress(): Boolean
+
+  def isMCGlobal(): Boolean
+
+  def isMCLinkLocal(): Boolean
+
+  def isMCNodeLocal(): Boolean
+
+  def isMCOrgLocal(): Boolean
+
+  def isMCSiteLocal(): Boolean
+
+  def isMulticastAddress(): Boolean
+
+  def isSiteLocalAddress(): Boolean
+
+}
+
+object InetAddress {
+
+  def getByName(host: String): InetAddress = {
+
+    if (host == null || host.length == 0)
+      return getLoopbackAddress()
+
+    var address: InetAddress = null
+    if (isValidIPv4Address(host)) {
+      val byteAddress: Array[Byte] = Array.ofDim[Byte](4)
+      val parts: Array[String]     = host.split("\\.")
+      val length: Int              = parts.length
+      if (length == 1) {
+        val value: Long = java.lang.Long.parseLong(parts(0))
+        for (i <- 0.until(4)) {
+          byteAddress(i) = (value >> ((3 - i) * 8)).toByte
+        }
+      } else {
+        for (i <- 0 until length) {
+          byteAddress(i) = java.lang.Integer.parseInt(parts(i)).toByte
+        }
+      }
+      if (length == 2) {
+        byteAddress(3) = byteAddress(1)
+        byteAddress(1) = 0
+      }
+      if (length == 3) {
+        byteAddress(3) = byteAddress(2)
+        byteAddress(2) = 0
+      }
+      address = new Inet4Address(byteAddress)
+    } else if (isValidIPv6Address(host)) {
+      var ipAddressString = host
+      if (ipAddressString.charAt(0) == '[') {
+        ipAddressString =
+          ipAddressString.substring(1, ipAddressString.length - 1)
+      }
+      val tokenizer: StringTokenizer =
+        new StringTokenizer(ipAddressString, ":.%", true)
+      val hexStrings            = new ArrayBuffer[String]()
+      val decStrings            = new ArrayBuffer[String]()
+      var scopeString: String   = null
+      var token: String         = ""
+      var prevToken: String     = ""
+      var prevPrevToken: String = ""
+      var doubleColonIndex: Int = -1
+      while (tokenizer.hasMoreTokens()) {
+        prevPrevToken = prevToken
+        prevToken = token
+        token = tokenizer.nextToken()
+        if (token.==(":")) {
+          if (prevToken.==(":")) {
+            doubleColonIndex = hexStrings.size
+          } else if (prevToken.!=("")) {
+            hexStrings.append(prevToken)
+          }
+        } else if (token.==(".")) {
+          decStrings.append(prevToken)
+        } else if (token.==("%")) {
+          if (prevToken.!=(":") && prevToken.!=(".")) {
+            if (prevPrevToken.==(":")) {
+              hexStrings.append(prevToken)
+            } else if (prevPrevToken.==(".")) {
+              decStrings.append(prevToken)
+            }
+          }
+          val buf: StringBuilder = new StringBuilder()
+          while (tokenizer.hasMoreTokens()) buf.append(tokenizer.nextToken())
+          scopeString = buf.toString
+        }
+      }
+      if (prevToken.==(":")) {
```

Comment:
The syntax is strange here and a few lines around. Is this code that has been generated? Would you mind rewriting it as `prevToken == ":"`?